libs/blocks/mobile-app-banner/mobile-app-banner.js

```
@@ -0,0 +1,72 @@
+  if (resp.ok) {
+    const json = await resp.json();
+    keyMatch = json.data.filter(
+      (p) => p.product === product,
+    );
+  }
+  return keyMatch[0]?.key;
+}
+
+/* eslint-disable */
+function branchInit(header, key) {
+  let initValue = false;
+  function initBranch() {
+    if (initValue) {
+      return;
+    }
+    initValue = true;
+    (function (b, r, a, n, c, h, _, s, d, k) {
+      if (!b[n] || !b[n]._q) {
+        for (; s < _.length;) c(h, _[s++]);
+        d = r.createElement(a);
+        // d.async = 1;
+        d.src = 'https://cdn.branch.io/branch-latest.min.js';
+        k = r.getElementsByTagName(a)[0];
+        k.parentNode.insertBefore(d, k);
+        b[n] = h;
+      }
+    })(
+      window,
+      document,
+      'script',
+      'branch',
+      function (b, r) {
+        b[r] = function () {
+          b._q.push([r, arguments]);
+        };
+      },
+      { _q: [], _v: 1 },
+      'addListener applyCode autoAppIndex banner closeBanner closeJourney creditHistory credits data deepview deepviewCta first getCode init link logout redeem referrals removeListener sendSMS setBranchViewData setIdentity track validateCode trackCommerceEvent logEvent disableTracking'.split(
+        ' '
+      ),
+      0
+    );
+    const privacyConsent = !!window.adobePrivacy && window.adobePrivacy.hasUserProvidedConsent();
+    branch.init(key, { tracking_disabled: !privacyConsent });
+    branch.addListener('didShowJourney', () => {
+      header.style.position = 'relative';
+    });
+    branch.addListener('didCloseJourney', () => {
+      header.style.position = 'sticky';
+    });
+  }
+  ['adobePrivacy:PrivacyConsent', 'adobePrivacy:PrivacyReject', 'adobePrivacy:PrivacyCustom']
+    .forEach((event) => {
+      window.addEventListener(event, initBranch);
+    });
+}
+
+/* eslint-enable */
+export default async function init(el) {
+  const header = document.querySelector('.global-navigation');
```

fail fast if the global navigation header is not available

---

This is done.